modules/cudacodec/include/opencv2/cudacodec.hpp

```diff
@@ -310,6 +329,23 @@ class CV_EXPORTS_W VideoReader
     /** @brief Returns information about video file format.
     */
     virtual FormatInfo format() const = 0;
+
+    /** @brief Signals VideoReader to start writing the raw bitstream to the given file from the next key frame.
+
+    @param filename Full path to the output file.  Each call to this with a new filename, filenameNew will wait for
+    the next key frame, stop writing to filename and start writing to filenameNew.  If VideoReader is already writing
+    to a file and filenameNew is empty, writing to filename will stop imidiately.
+    @param autoDetectExt Automatically detect and append the codec extension to filename before writing.
+
+    The function is intended to be used when streaming from an RTSP source where it is either
+    a) desirable to store the original streamed data, or
+    b) the overhead of re-encoding the decoded frame is undesirable.
+
+    @note This should only be used when streaming from rtsp and is not guaranteed to work when reading from a
+    file, especially from container formats avi, mp4 etc.  If the filename provided is invalid, cannot be opened
+    or written to, the first call to nextFrame() after calling this function will return false.
+     */
+    CV_WRAP virtual void writeToFile(const std::string filename, const bool autoDetectExt = false) = 0;
```

Comment:
Thank you for taking a look.

> Be careful with creation of GOD classes.

I am not sure how to achieve my objective without creating a GOD class.  I want to be able to stream and decode from an RTSP source at the same time as archiving the footage.   This seems like a common use case for IP camera's, e.g. run DNN over live decoded footage and archive footage for training later or to simply see why the DNN failed.
With two classes I would need two streams which is what I am trying to avoid due to unecessary network overhead and lack of support from IP camera's (usually each stream is a different resolution if multiple streams are supported).
Can you suggest a way to achieve this without placing the read and write functionality in the same class?